jcl/src/java.base/share/classes/com/ibm/jit/JITHelpers.java

```diff
@@ -69,6 +70,49 @@ private static JITHelpers jitHelpers() {
 		return helpers;
 	}
 
+	/*
+	 * Public methods to provide access to expected exception
+	 */
+
+	public static void setExpectedException(Class<? extends Throwable> expectedException) {
+		Thread currentThread = Thread.currentThread();
+		String vmThreadName = currentThread.getName();
+		initExceptionMap();
+		
+		if (expectedException == null) {
+			System.out.println("Removing exception for thread: " + vmThreadName);
+			exceptionMap.remove(vmThreadName);
+		}
+		else {
+			System.out.println("Setting expected exception for " + vmThreadName);
+			exceptionMap.put(vmThreadName, expectedException);
+		}
+	}
+	public static Class<? extends Throwable> getExpectedException() {
+		Thread currentThread = Thread.currentThread();
+		if(exceptionMap == null) {
+			return null;
+		}
+		return exceptionMap.get(currentThread.getName());
+	}
+
+	/*
+	 * Public methods to instantiate and destroy exceptionMap
+	 */
+
+	public static void initExceptionMap() {
+		if (exceptionMap == null) {
+			exceptionMap = new java.util.HashMap<String, Class<? extends Throwable>>();
```

Comment:
We should probably synchronize this method on `JitHelpers.class` since multiple threads can be initializing the map.

Reply:
@r30shah @fjeremic 
I believe this may have been causing the issue with the wrong method being picked up. When I tested it with synchronization it went away. Ref: https://github.com/r30shah/openj9-jit-debug-agent/issues/5#issuecomment-973206780
[non-working-example.log](https://github.com/r30shah/openj9-jit-debug-agent/files/7573901/non-working-example.log)
[working-example.log](https://github.com/r30shah/openj9-jit-debug-agent/files/7573902/working-example.log)